code/modules/projectiles/guns/energy.dm

```diff
@@ -593,4 +548,346 @@
+	reload_sound = 'sound/weapons/guns/interact/standard_laser_pistol_reload.ogg'
+	fire_sound = 'sound/weapons/guns/fire/Laser Pistol Standard.ogg'
+	force = 20
+	icon_state = "tep"
+	item_state = "tep"
+	icon = 'icons/Marine/gun64.dmi'
+	w_class = WEIGHT_CLASS_NORMAL
+	flags_equip_slot = ITEM_SLOT_BELT
+	max_shots = 30 //codex stuff
+	load_method = CELL //codex stuff
+	ammo = /datum/ammo/energy/lasgun/marine
+	ammo_diff = null
+	cell_type = /obj/item/cell/lasgun/lasrifle/marine
+	charge_cost = 20
+	gun_firemode = GUN_FIREMODE_AUTOMATIC
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO, GUN_FIREMODE_AUTOMATIC)
+
+	attachable_allowed = list(
+		/obj/item/attachable/bayonet,
+		/obj/item/attachable/bayonetknife,
+		/obj/item/attachable/reddot,
+		/obj/item/attachable/lasersight,
+		/obj/item/attachable/flashlight,
+		/obj/item/attachable/scope/mini,
+	)
+
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_AMMO_COUNTER|GUN_ENERGY|GUN_AMMO_COUNTER
+	attachable_offset = list("muzzle_x" = 23, "muzzle_y" = 22,"rail_x" = 12, "rail_y" = 22, "under_x" = 16, "under_y" = 14, "stock_x" = 22, "stock_y" = 12)
+
+	aim_slowdown = 0.2
+	wield_delay = 0.6 SECONDS
+	scatter = 0
+	scatter_unwielded = 0
+	fire_delay = 0.2 SECONDS
+	accuracy_mult = 1.1
+	accuracy_mult_unwielded = 0.9
+	scatter_unwielded = 0
+	mode_list = list(
+		"Standard" = /datum/lasrifle/base/energy_pistol_mode/standard,
+		"Heat" = /datum/lasrifle/base/energy_pistol_mode/heat,
+		"Disabler" = /datum/lasrifle/base/energy_pistol_mode/disabler,
+	)
+
+/datum/lasrifle/base/energy_pistol_mode/standard
+	charge_cost = 20
+	ammo = /datum/ammo/energy/lasgun/marine/pistol
+	fire_delay = 0.25 SECONDS
+	fire_sound = 'sound/weapons/guns/fire/Laser Pistol Standard.ogg'
+	message_to_user = "You set the laser pistol's charge mode to standard fire."
+	fire_mode = GUN_FIREMODE_AUTOMATIC
+	icon_state = "tep"
+
+/datum/lasrifle/base/energy_pistol_mode/disabler
+	charge_cost = 80
+	ammo = /datum/ammo/energy/lasgun/marine/pistol/disabler
+	fire_delay = 10
+	fire_sound = 'sound/weapons/guns/fire/disabler.ogg'
+	message_to_user = "You set the laser pistol's charge mode to disabler fire."
+	fire_mode = GUN_FIREMODE_AUTOMATIC
+	icon_state = "tep"
+	radial_icon_state = "laser_disabler"
+
+/datum/lasrifle/base/energy_pistol_mode/heat
+	charge_cost = 110
+	ammo = /datum/ammo/energy/lasgun/marine/pistol/heat
+	fire_delay = 0.5 SECONDS
+	fire_sound = 'sound/weapons/guns/fire/laser3.ogg'
+	message_to_user = "You set the laser pistol's charge mode to wave heat."
+	fire_mode = GUN_FIREMODE_AUTOMATIC
+	icon_state = "tep"
+	radial_icon_state = "laser_heat"
+
+//TE Standard Laser Carbine
+
+/obj/item/weapon/gun/energy/lasgun/lasrifle/standard_marine_carbine
+	name = "\improper Terra Experimental laser carbine"
+	desc = "A TerraGov standard issue laser carbine otherwise known as TE-C abbreviated has an integrated charge selector for burst and scatter settings. Uses standard Terra Experimntal, TE abbreviated power cells."
```

Comment:
```suggestion
	desc = "A TerraGov standard issue laser carbine otherwise known as TE-C for short. It has an integrated charge selector for burst and scatter settings. Uses standard Terra Experimental (abbreviated as TE) power cells."
```